Metadata docs: Update expected structure (#325)
aragon/court-dashboard: docs/metadata.md

}
```

In the frontend we try to parse the `metadata` as a JSON object.

### metadata - JSON object

We are going to try to find the following keys:

1. `description`
2. `metadata`

> If `metadata` is a valid IPFS hash/cid we'll fetch its content.

Reviewer: ```suggestion
If `metadata` is a valid [native IPFS path](https://docs.ipfs.io/guides/guides/addressing/#native-urls) we'll fetch its content through IPFS.
```

Wanted to double check, we are assuming the `ipfs://` protocol prefix, right (although we still attempt to fallback to detecting if it's a CID/IPLD path if it doesn't exist)?

Author: Oh yeah good catch.

Author: We expect to be in the form of `ipfs:{cid | hash}`. Not sure if it's the same?

Reviewer: Ah shit, no it isn't :( (it needs the `://`, not just the `:`).

Nevermind, I've changed the suggestion to something else!

}
```

#### metadata - String

> In case `metadata` is a plain string, we assume it to be the dispute's description and set the dispute creator (`subject.id`) as the plaintiff. No agreement link will be available.

Author: Note that we only require the `metadata` field to be A JSON object. If we can't parse it, we are displaying the `Could not fetch content from IPFS` error.

> > Note that even though the description is present in the ipfs content, we require to also be present at the `metdata` field. The reason for this is so we don't need to fetch all disputes ipfs content when loading all disputes cards.

Reviewer: ```suggestion
> Note that even though `description` should be present in the IPFS content, we also assume it to be present in the `metadata` field. We do this to avoid needing to fetch every disputes' IPFS content when loading their dispute cards.
```

}
```

> The dashboard checks if the `data` field is a valid IPFS hash/cid

Reviewer: Should this also have the same correction above, as a native IPFS URL?

### data - IPFS hash

>   Try to fetch the content from IPFS

Reviewer: It looks like we may have added some additional indentation here.

> ### data - IPFS string

Reviewer: ```suggestion
### data - IPFS path
```
